tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -122,5 +126,90 @@ public void ChangeStreamExample3()
                 next.FullDocument.Should().Be(documents[1]);
             }
         }
+
+        [Fact]
+        public void Changestream_Example_4()
+        {
+            RequireServer.Check().Supports(Feature.AggregateAddFields);
+
+            var client = DriverTestConfiguration.Client;
+            var database = client.GetDatabase("ChangeStreamExamples");
+            database.DropCollection("inventory");
+            var inventoryCollection = database.GetCollection<BsonDocument>("inventory");
+
+            //const pipeline = [
+            //{ $match: { $or : [{ 'fullDocument.username': 'alice' }, { 'operationType' : 'delete' }] } },
+
+            //{ $addFields: { newField: 'this is an added field!' } }
+
+            //];
+            //const collection = db.collection('inventory');
+            //const changeStream = collection.watch(pipeline);
+            //const next = await changeStream.next();
+
+            var cancelationTokenSource = new CancellationTokenSource();
+            try
+            {
+                BsonDocument document = new BsonDocument("username", "alice");
+
+                Task.Run(() =>
+                {
+                    while (!cancelationTokenSource.IsCancellationRequested)
+                    {
+                        Thread.Sleep(TimeSpan.FromMilliseconds(500));
```

Comment:
Bump this down to 50 or 100 ms so that the test runs faster.

Reply:
Note: in case Task.Run will throw any exception by any reason until enumerator.MoveNext() will be passed, MoveNext method will infinitely wait for the first change. As I see, previous examples have the same problem. Should we consider this potential problem as a bug?